File: frontend/src/microservices/attraction/objects/AttractionTicketSessionResponse.ts
// 本文件定义 attraction 模块的 `AttractionTicketSessionResponse`，作为响应数据并提供 JSON 编解码。

export type AttractionTicketSessionResponse = {
  sessionId: string
  sessionName: string
  useDate: string
  startsAt: string
  endsAt: string
  capacity: number
  availableQuantity: number | null
  status: string
}
export const attractionTicketSessionResponseFromJson = (json: string): AttractionTicketSessionResponse =>
  JSON.parse(json) as AttractionTicketSessionResponse

export const attractionTicketSessionResponseToJson = (value: AttractionTicketSessionResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/AttractionTicketSessionResponse.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AttractionTicketSessionResponse(
    sessionId: String,
    sessionName: String,
    useDate: String,
    startsAt: String,
    endsAt: String,
    capacity: Int,
    availableQuantity: Int,
    status: String
)
object AttractionTicketSessionResponse:
  given sourceEncoder: Encoder[AttractionTicketSessionResponse] = deriveEncoder
  given sourceDecoder: Decoder[AttractionTicketSessionResponse] = deriveDecoder
